segment/writer.go

```diff
@@ -539,3 +547,28 @@ func (w *Writer) Sealed() (bool, uint64, error) {
 func (w *Writer) LastIndex() uint64 {
 	return atomic.LoadUint64(&w.commitIdx)
 }
+
+// SegmentInfo returns writer's atomic info field
+func (w *Writer) SegmentInfo() types.SegmentInfo {
+	return w.info.Load().(types.SegmentInfo)
```

Comment:
I did not check for the assertion error here as the only reader of the field is `types.SegmentInfo`. Please let me know if you feel I should add that error check.